src/webui/www/private/scripts/client.js

```diff
@@ -581,14 +585,23 @@ window.addEvent('load', function() {
                 }
                 clearTimeout(syncMainDataTimer);
                 syncMainDataTimer = syncMainData.delay(getSyncMainDataInterval());
+                requestInProgress = false;
+                $('syncing_div').set('html', '');
             }
-        }).send();
+        });
+        requestInProgress = true;
+        request.send();
     };
 
     updateMainData = function() {
         torrentsTable.updateTable();
-        clearTimeout(syncMainDataTimer);
-        syncMainDataTimer = syncMainData.delay(100);
+        if (!requestInProgress){
+            clearTimeout(syncMainDataTimer);
+            syncMainDataTimer = syncMainData.delay(100);
+        }
+        else {
+            $('syncing_div').set('html', 'Syncing...');
+        }
```

Comment:
`syncMainData.delay` seems to get called in a few places, including in `showTransfersTab()`. These calls circumvent the `requestInProgress` logic.

I propose we move the highlighted logic to a new `syncData(delay)` function, and replace all `clearTimeout` and `syncMainData.delay` calls with calls to `syncData`. This way, `syncData` is the only place that calls `syncMainData.delay`, which also reduces the likelihood that someone else will accidentally circumvent the `requestInProgress` logic in the future.

Reply:
Ok, I implemented this change